src/trace/entry.zig

```diff
@@ -1,217 +1,236 @@
-        message.appendSlice(msg[0..]) catch @panic("could not appendSlice for message");
-        self.message = message;
-        self.time = time.DateTime.now();
-        self.level = level;
-        self.channel.send(self) catch @panic("could not send to channel");
+        return buf;
     }
+};
 
-    pub fn format(
-        self: *const Self,
-        _: []const u8,
-        _: std.fmt.FormatOptions,
-        writer: anytype,
-    ) !void {
-        // default formatting style
-        try logfmt.formatter(self, writer);
+/// An instance of `Entry` that logs directly to std.debug.print, instead of sending to channel.
+pub const StdErrEntry = struct {
+    const builtin = @import("builtin");
+
+    const Self = @This();
+    log_level: Level,
+    scope: ?[]const u8,
+    allocator: std.mem.Allocator,
+    log_msg: ?std.ArrayList(u8),
+
+    pub fn init(
+        allocator: std.mem.Allocator,
+        scope: ?[]const u8,
+        log_level: Level,
+    ) Self {
+        return .{
+            .allocator = allocator,
+            .scope = scope,
+            .log_level = log_level,
+            .log_msg = std.ArrayList(u8).init(allocator),
+        };
     }
 
-    pub fn custom_format(self: *const Self, formatter: anytype, writer: anytype) !void {
-        try formatter(self, writer);
+    pub fn deinit(self: *Self) void {
+        if (self.log_msg) |log_msg| {
+            log_msg.deinit();
+        }
     }
-};
 
-const A = enum(u8) {
-    some_enum_variant,
-};
+    pub fn field(self: *Self, name: []const u8, value: anytype) *Self {
+        logfmt.fmtField(self.log_msg.?.writer(), name, value);
+        return self;
+    }
 
-test "trace.entry: should info log correctly" {
-    var logger = Logger.init(testing.allocator, Level.info);
-    defer logger.deinit();
-    var entry = StandardEntry.init(testing.allocator, logger.standard.channel, .debug);
-    defer entry.deinit();
-
-    const anull: ?u8 = null;
-
-    entry
-        .field("some_val", true)
-        .field("enum_field", A.some_enum_variant)
-        .field("name", "a-mod")
-        .field("elapsed", @as(i48, 135133340042))
-        .field("possible_value", anull)
-        .logf(.info, "hello, {s}", .{"world!"});
-}
+    pub fn log(self: *Self, msg: []const u8) void {
+        defer self.deinit();
+        const log_msg = logfmt.LogMsg{
+            .level = self.log_level,
+            .maybe_scope = self.scope,
+            .maybe_msg = msg,
+            .maybe_fields = null,
+            .maybe_fmt = null,
+        };
+
+        const writer = self.log_msg.?.writer();
+        logfmt.writeLog(writer, log_msg) catch @panic("Failed to write log");
+        std.debug.print("{?s}", .{self.log_msg.?.items});
+    }
+
+    pub fn logf(self: *Self, comptime fmt: []const u8, args: anytype) void {
+        defer self.deinit();
+        // Format message.
+        var fmt_msg = std.ArrayList(u8).initCapacity(self.allocator, 256) catch @panic("could not initCapacity for message");
```

Comment:
This is intended to be used for testing (it was named `TestEntry` before being renamed to `StdErrEntry`) so yeah it can be hard coded to a reasonable value and drop the allocation.